modules/thermo-cycler/thermo-cycler-arduino/fan.cpp

```diff
@@ -0,0 +1,49 @@
+#include "fan.h"
+#include "high_frequency_pwm.h"
+
+Fan::Fan()
+{}
+
+void Fan::set_percentage(float p)
+{
+  enable();
+  uint8_t val = p * 255.0;
+  if (val < 0)
+  {
+    val = 0;
+  }
+  else if (val > 255)
+  {
+    val = 255;
+  }
+  current_power = p;
+  hfq_analogWrite(_pwm_pin, val);
+}
+
+void Fan::setup_pwm_pin(uint8_t pwm_pin)
+{
+  _pwm_pin = pwm_pin;
+  _pwm_controlled = true;
```

Comment:
Is there any way to switch off pwm?

Reply:
Nope. Not in this code. The "switching off" is basically achieved by setting a pwm value of 0

Reply:
👍